fs/vfs/fs_dupfd.c

```diff
@@ -73,7 +73,7 @@ int file_dup(FAR struct file *filep, int minfd)
   fd2 = files_allocate(NULL, 0, 0, minfd);
   if (fd2 < 0)
     {
-      return -EMFILE;
+      return fd2;
```

Comment:
does `files_allocate` return `EMFILE` somehow?
The return of `file_dup` is the value that we gonna set `errno` to.

Reply:
Yes, see here:
https://github.com/apache/incubator-nuttx/pull/2639/files#diff-1fafb74ef7557187f97e4ffa7b8d30c3e6edc1df89cc952ecab48fdbe86e637fL334

Reply:
Ah right!  Thanks.  I've been looking at my source tree when I asked that.